main.py

```diff
@@ -196,21 +197,19 @@ def tmp(_, inp, out):
                     spqr_handlers[name].add_batch(inp[0].data)
 
                 return tmp
-
             handles = []
             for sublayer_name in subset:
                 handles.append(subset[sublayer_name].register_forward_hook(add_batch(sublayer_name)))
             for j in trange(
                 args.nsamples, desc="calc outs before quantization", leave=False
-            ):
-                outs[j] = layer(inps[j].unsqueeze(0), **forward_args)[0]
+            ):  
+                outs[j] = layer(inps[j].to(layer_dev).unsqueeze(0), **forward_args)[0]
```

Comment:
check that inps[j] is on cpu

Reply:
checked. it stays on cpu indeed.  `.to()` creates copy.